src/main/java/org/mule/tools/devkit/sonar/checks/java/NoNullAnnotationsInProcessorCheck.java

```diff
@@ -0,0 +1,51 @@
+package org.mule.tools.devkit.sonar.checks.java;
+
+import com.google.common.collect.Iterables;
+import org.jetbrains.annotations.NotNull;
+import org.mule.tools.devkit.sonar.utils.ClassParserUtils;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.tree.*;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+
+import java.util.List;
+
+@Rule(key = NoNullAnnotationsInProcessorCheck.KEY, name = "No @Nullable/@NotNull annotations should be used at processor level", description = "If there is a parameter annotated with @NotNull (or @NonNull), remove the annotation. If there is a parameter annotated with @Nullable, replace it with @Optional annotation or remove it if @Optional is already present.", priority = Priority.CRITICAL, tags = { "connector-certification" })
+@ActivatedByDefault
+public class NoNullAnnotationsInProcessorCheck extends AbstractConnectorClassCheck {
+
+    public static final String KEY = "nullable-and-notnull-annotations";
+
+    @Override
+    protected void verifyProcessor(@NotNull MethodTree tree, @NotNull final IdentifierTree processorAnnotation) {
+        for (VariableTree var : tree.parameters()) {
+            final List<AnnotationTree> annotations = var.modifiers().annotations();
+
+            int indexOfNullable = Iterables.indexOf(annotations, ClassParserUtils.hasSimpleAnnotationPredicate("Nullable"));
```

Comment:
Shouldn't we check for multiple providers of nullcheck annotations, beside jetbrains?